src/payloads/send_animation.rs

```diff
@@ -0,0 +1,52 @@
+// This file is auto generated by `cg` <https://github.com/teloxide/cg> (e634f65).
+// **DO NOT EDIT THIS FILE**,
+// edit `cg` instead.
+use serde::Serialize;
+
+use crate::types::{ChatId, InputFile, Message, ParseMode, ReplyMarkup};
+
+impl_payload! {
+    /// Use this method to send animation files (GIF or H.264/MPEG-4 AVC video without sound). On success, the sent [`Message`] is returned. Bots can currently send animation files of up to 50 MB in size, this limit may be changed in the future.
+    ///
+    /// [`Message`]: crate::types::Message
+    #[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize)]
+    pub SendAnimation (SendAnimationSetters) => Message {
+        required {
+            /// Unique identifier for the target chat or username of the target channel (in the format `@channelusername`)
+            pub chat_id: ChatId [into],
+            /// Animation to send. Pass a file_id as String to send a video that exists on the Telegram servers (recommended), pass an HTTP URL as a String for Telegram to get a video from the Internet, or upload a new video using multipart/form-data. [More info on Sending Files »]
+            ///
+            /// [More info on Sending Files »]: crate::types::InputFile
+            pub animation: InputFile,
+            /// Animation caption (may also be used when resending videos by _file\_id_), 0-1024 characters after entities parsing
+            pub caption: String [into],
+        }
+        optional {
+            /// Duration of the animation in seconds
+            pub duration: u32,
+            /// Animation width
+            pub width: u32,
+            /// Animation height
+            pub height: u32,
+            /// Thumbnail of the file sent; can be ignored if thumbnail generation for the file is supported server-side. The thumbnail should be in JPEG format and less than 200 kB in size. A thumbnail's width and height should not exceed 320. Ignored if the file is not uploaded using multipart/form-data. Thumbnails can't be reused and can be only uploaded as a new file, so you can pass “attach://<file_attach_name>” if the thumbnail was uploaded using multipart/form-data under <file_attach_name>. [More info on Sending Files »]
+            ///
+            /// [More info on Sending Files »]: crate::types::InputFile
+            pub thumb: InputFile,
+            /// Mode for parsing entities in the animation caption. See [formatting options] for more details.
+            ///
+            /// [formatting options]: https://core.telegram.org/bots/api#formatting-options
+            pub parse_mode: ParseMode,
+            /// Sends the message [silently]. Users will receive a notification with no sound.
+            ///
+            /// [silently]: https://telegram.org/blog/channels-2-0#silent-messages
+            pub disable_notification: bool,
+            /// If the message is a reply, ID of the original message
+            pub reply_to_message_id: i64,
```

Comment:
`i64` -> `i32`

Reply:
Ah, right.

But while we at it: can we change message-id to be u32? In practice, it always starts at 0 and is monotonically increasing.

Reply:
> In practice, it always starts at 0 and is monotonically increasing.

Could you please cite the official documentation? I can't find these words, especially that `message_id` always starts at 0.

Reply:
As I've said
>In practice

The telegram docs are garbage and don't say anything.

Reply:
But what if they returned -1, for example, to indicate something?

Reply:
Though the message-id can only be obtained from the telegram itself, so the type doesn't really matter. I'll fix this.